src/main/java/seedu/toluist/controller/HistoryController.java

```diff
@@ -18,19 +18,19 @@
     private static final String COMMAND_WORD = "history";
     private static final String COMMAND_REGEX = "(?iu)^\\s*history\\s*";
 
-    private ArrayList<String> commandHistory;
+    private CommandHistoryList commandHistory;
 
-    public void setCommandHistory(ArrayList<String> commandHistory) {
+    public void setCommandHistory(CommandHistoryList commandHistory) {
         this.commandHistory = commandHistory;
     }
 
     public void execute(String command) {
         logger.info(getClass().getName() + " will handle command");
 
-        String result = String.join("\n", commandHistory);
+        String result = String.join("\n", commandHistory.getAsArrayList());
 
         uiStore.setCommandResult(new CommandResult(String.format(
-                RESULT_MESSAGE, result, StringUtil.nounWithCount("command", commandHistory.size()))));
+                RESULT_MESSAGE, result, StringUtil.nounWithCount("command", commandHistory.getAsArrayList().size()))));
```

Comment:
commandHistory.getAsArrayList() is called twice so can consider making it a variable first.
List<String> commandHistoryList = commandHistory.get.....